src/emulator/kernel/src/load_self.cpp

```diff
@@ -56,12 +59,63 @@ static bool load_var_imports(const uint32_t *nids, const Ptr<uint32_t> *entries,
             const char *const name = import_name(nid);
             LOG_DEBUG("\tNID {:#08x} ({}) at {:#x}", nid, name, entry.address());
         }
+
+        uint32_t *const stub = entry.get(mem);
+
+        if(nid == NID_STACK_CHK_GUARD){
+            stub[0] = __stack_chk_guard;
+            continue;
+        }
+
+        const ExportNids::iterator export_address = kernel.export_nids.find(nid);
+        if (export_address == kernel.export_nids.end()) {
+            const char *const name = import_name(nid);
+            LOG_ERROR("\tNID NOT FOUND {:#08x} ({}) at {:#x}: {:#08x}", nid, name, entry.address(), stub[0]);
+            continue;
+        }
+        uint32_t *const export_ptr = Ptr<uint32_t>(export_address->second).get(mem);
+        stub[0] = export_ptr[0];
     }
 
     return true;
 }
 
-static bool load_func_imports(const uint32_t *nids, const Ptr<uint32_t> *entries, size_t count, const MemState &mem) {
+// Encode code taken from https://github.com/yifanlu/UVLoader/blob/master/resolve.c
+
+#define INSTRUCTION_UNKNOWN     0       ///< Unknown/unsupported instruction
+#define INSTRUCTION_MOVW        1       ///< MOVW Rd, \#imm instruction
+#define INSTRUCTION_MOVT        2       ///< MOVT Rd, \#imm instruction
+#define INSTRUCTION_SYSCALL     3       ///< SVC \#imm instruction
+#define INSTRUCTION_BRANCH      4       ///< BX Rn instruction
+
+static uint32_t encode_arm_inst (uint8_t type, uint16_t immed, uint16_t reg) {
+    switch(type)
```

Comment:
This switch is against current clang-format rules we use, if you want to hold that, you should add a `//clang-format off`